lib/ansible/galaxy/collection/galaxy_api_proxy.py

```diff
@@ -85,13 +88,19 @@ def get_collection_versions(self, requirement):
         # type: (Requirement) -> Iterable[Tuple[str, GalaxyAPI]]
         """Get a set of unique versions for FQCN on Galaxy servers."""
         if requirement.is_concrete_artifact:
-            return {
-                (
-                    self._concrete_art_mgr.
-                    get_direct_collection_version(requirement),
-                    requirement.src,
-                ),
-            }
+            version = self._concrete_art_mgr.get_direct_collection_version(requirement)
+
+            err = f"Invalid version found for requirement '{requirement}': {version} ({type(version)}). "
+            err += "The version should be a semantic version string (or '*')."
+            if not isinstance(version, string_types):
+                raise AnsibleError(err)
+            elif version != "*":
+                try:
+                    SemanticVersion(version)
+                except ValueError:
+                    raise AnsibleError(err)
+
+            return {(version, requirement.src,)}
```

Comment:
I think it looks slightly nicer w/o that comma
```suggestion
            return {(version, requirement.src)}
```